### PR TITLE
refactor: make NodeList an actual list

### PR DESCRIPTION
**Pull request by simPod**
Requires https://github.com/webonyx/graphql-php/pull/1314

### CODE REVIEW

File: src/Language/Visitor.php [hunk 1]
```diff
@@ -249,9 +253,13 @@ public static function visit(object $root, array $visitor, ?array $keyMap = null
                     : null;
                 $node = $parent !== null
                     ? (
-                        $parent instanceof NodeList || \is_array($parent)
-                            ? $parent[$key]
-                            : $parent->{$key}
+                        $parent instanceof NodeList
+                            ? $parent->get($key)
+                            : (
+                                \is_array($parent)
```
Comment (spawnia) [inline src/Language/Visitor.php:259]: Is this ever the case?

**Review by spawnia — COMMENTED** [review]
Really like the direction of this.

File: src/Language/Visitor.php (hunk 1 above)
Comment (simPod) [reply src/Language/Visitor.php]: No test touches it so I removed it.

File: src/Language/Visitor.php [hunk 2]
```diff
@@ -222,12 +222,13 @@ public static function visit(object $root, array $visitor, ?array $keyMap = null
...
-                            if ($node instanceof NodeList || \is_array($node)) {
-                                $node[$editKey] = $editValue;
-                            } else {
-                                $node->{$editKey} = $editValue;
+                        } elseif ($node instanceof NodeList) {
+                            if ($editValue !== false) {
+                                $node->set($editKey, $editValue);
                             }
+                        } else {
+                            \assert(! \is_array($node));
```
[the remark this answers is not in the document]
Comment (simPod) [reply src/Language/Visitor.php]: Good catch, I could clean that up.

**Comment by spawnia** [comment]
I just fixed and merged https://github.com/webonyx/graphql-php/pull/1314.

**Pull request merged**